crypto/asn1/asn_mime.c

```diff
@@ -791,29 +791,18 @@ static MIME_HEADER *mime_hdr_new(const char *name, const char *value)
 {
     MIME_HEADER *mhdr = NULL;
     char *tmpname = NULL, *tmpval = NULL, *p;
-    int c;
 
     if (name) {
         if ((tmpname = OPENSSL_strdup(name)) == NULL)
             return NULL;
-        for (p = tmpname; *p; p++) {
-            c = (unsigned char)*p;
-            if (isupper(c)) {
-                c = tolower(c);
-                *p = c;
-            }
-        }
+        for (p = tmpname; *p; p++)
+            *p = ossl_tolower(*p);
```

Comment:
Here is by the way "fun" question in the context. Why is it lowered? I don't actually know, but it's likely because string comparison elsewhere is specified to be case insensitive. Are non-ASCII characters permitted here? And if so, are they supposed to be case insensitive? If so, then it should remain locale-specific... But even worse what if this string is UTF-8 with non-ASCII characters? Maybe this is all wrong and we would be better off using strcasecmp instead of lowering strings. Or if non-ASCII characters are actually not permitted, then we would be better off rejecting such strings and returning error...